jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/RawConfiguration.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.plugins.common;
+
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Holds raw plugin configuration parameter values. Acts as a common adapter for heterogeneous
+ * plugin configuration models.
+ */
+public interface RawConfiguration {
+
+  @Nullable
+  String getFromImage();
+
+  AuthProperty getFromAuth();
+
+  @Nullable
+  String getFromCredHelper();
+
+  Iterable<String> getToTags();
+
+  @Nullable
+  List<String> getEntrypoint();
+
+  @Nullable
+  List<String> getProgramArguments();
+
+  @Nullable
+  String getMainClass();
+
+  List<String> getJvmFlags();
+
+  String getAppRoot();
+
+  Map<String, String> getEnvironment();
+
+  Map<String, String> getLabels();
+
+  List<String> getPorts();
+
+  @Nullable
+  String getUser();
+
+  boolean getUseCurrentTimestamp();
+
+  boolean getAllowInsecureRegistries();
+
+  boolean getUseOnlyProjectCache();
+
+  @Nullable
+  AuthProperty getInferredAuth(String authTarget) throws InferredAuthRetrievalException;
```

Comment:
Yeah, it also makes sense to feed it to the config processor rather than having it in the raw configuration, and I have thought about that too, but after some thinking, I got to the conclusion that this auth info is just values from a static file (`settings.xml`) not much different from getting values from `pom.xml`, and conceptually fits with "raw configuration values". However, I also see we may need to decrypt the values, so in another sense, it requires some processing. I'll add a TODO for considering passing an additional base image auth provider into the config processor.